scripts/events/test_native_event_architecture.sh

```
@@ -0,0 +1,143 @@
+grep -Fq 'filter->owner_generation_set = false;' "$project_root/src/event_debug.c" ||
+  fail "entity filtering no longer spans all live generations for the selected owner"
+grep -Fq 'Payloads: redacted' "$project_root/src/event_debug.c" ||
+  fail "eventdebug payload redaction is missing"
+
+# These linked CuTests supply runtime evidence for the source constraints above.
+for test_name in \
+  Test_event_runtime_profiles_native_semantic_callbacks \
+  Test_event_runtime_owner_cancel_invalidates_handle_and_cleans_once \
+  TestActiveWorldDormantPopulationDoesNotCreateScheduledWork \
+  Test_event_debug_registry_is_backend_neutral_filterable_and_width_bounded; do
+  grep -REq --include='*.c' "void $test_name[[:space:]]*\(" \
```

Shellcheck (0.11.0)

[error] Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet).

(SC1087)